### PR TITLE
Addon-Docs: Do not create extra Vue instance for Dynamic source rendering

### LINKED ISSUE #13633: Duplicated mounted call on component in Vue story
**Describe the bug**
When I include a component in a story, the 'mounted' lifecycle is called twice. This doesn't happen when the component is used outside Storybook

**Expected behavior**
I expect that the 'mounted' lifecycle is called only once

**Code snippets**
This is my code. I'm following the structure tu use 'CONTROLS'

```import StarRating from './index.vue';
import Docs from './index.mdx';
import { action } from '@storybook/addon-actions';

export default { title: '2. Molecules/StarRating', component: StarRating };

// Parameters of the story
const parametersStory = {
  controls: { expanded: true, hideNoControlsWarning: true },
  docs: { page: Docs },
  design: { disable: true },
  jest: ['starRating.test.js']
};

// Argtypes for controls
const argTypesStory = {
  skillLevel: { control: { disable: true } },
  levels: { control: { disable: true } }
};

// Actions on events
const actionsData = {
  onLevelSelected: action('onLevelSelected')
};

// The template with args and argTypes
const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { StarRating },
  template:
    '<StarRating\n' +
    ' :disabled="disabled"' +
    ' :visibleEmptyStar="visibleEmptyStar"' +
    ' :levels="levels"\n' +
    ' :skillLevel="skillLevel"\n' +
    ' @onLevelSelected="onLevelSelected"\n' +
    '></StarRating>'
});

// The Default component that extends the Template
export const Default = Template.bind({});
Default.args = Object.assign(
  {},
  {
    levels: [1, 2, 3, 4, 5],
    skillLevel: 2
  },
  actionsData
);

// Final declarations to use the story
Default.parameters = parametersStory;
Default.argTypes = argTypesStory;
```

In the 'Template' declaration, if I didn't include the part `args, { argTypes }` it mounts only once, as expected. However, without that code, controls don't work.

**Version**
I'm using the Storybook version 6.1.14 and the addons have all the same version

Could you please help me to find a solution to this struggling problem? Thanks a lot

### PR DESCRIPTION
**Pull request by pocka**
Issue: #13633 (already closed though)

Reduce the amount of the work for #13862
Probably related to #13697 and #13721 (hopefully this patch will fix them)

## What I did

Removed `new Vue` in source decorator, Added decorator component that renders story source code based on VNode.

## How to test

- Is this testable with Jest or Chromatic screenshots? ... No
- Does this need a new example in the kitchen sink apps? ... No
- Does this need an update to the documentation? ... No

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

File: addons/docs/src/frameworks/vue/sourceDecorator.ts [hunk 1]
```diff
@@ -30,47 +30,45 @@ export const sourceDecorator = (storyFn: any, context: StoryContext) => {
...
 
-  return story;
+  return {
+    components: {
+      Story: story,
+    },
+    // We need to wait until the wrapper component to be mounted so Vue runtime
+    // struct VNode tree. We get `this._vnode == null` if switch to `created`
+    // lifecycle hook.
+    mounted() {
```
Comment (pocka) [inline addons/docs/src/frameworks/vue/sourceDecorator.ts:44]: It seems every control change triggers re-mount so no need to use `updated` hook?

File: addons/docs/src/frameworks/vue/sourceDecorator.ts [hunk 2]
```diff
@@ -30,47 +30,45 @@ export const sourceDecorator = (storyFn: any, context: StoryContext) => {
...
+            // Because the parsed vnode missing spaces right before/after the surround tag,
+            // we always get weird wrapped code without this option.
+            htmlWhitespaceSensitivity: 'ignore',
+          })
+        );
+      } catch (e) {
+        logger.warn(`Failed to generate dynamic story source: ${e}`);
+      }
+    },
+    template: '<story />',
```
Comment (pocka) [inline addons/docs/src/frameworks/vue/sourceDecorator.ts:70]: Should I write in this form?

```diff
 return {
- components: {
-    Story: story,
-  },
   mounted() {
     // ...
   },
-  template: '<story />',
+  render(h) { return h(story) },
 }
```

File: addons/docs/src/frameworks/vue/sourceDecorator.ts (hunk 1 above)
Comment (phated) [reply addons/docs/src/frameworks/vue/sourceDecorator.ts:44]: Yeah, I believe that is the current implementation.

File: addons/docs/src/frameworks/vue/sourceDecorator.ts (hunk 2 above)
Comment (phated) [reply addons/docs/src/frameworks/vue/sourceDecorator.ts:70]: What's the difference?

**Review by phated — APPROVED** [review]
Trying to re-run the yarn pnp e2e tests, but otherwise this seems to make sense to me.

File: addons/docs/src/frameworks/vue/sourceDecorator.ts (hunk 2 above)
Comment (pocka) [reply addons/docs/src/frameworks/vue/sourceDecorator.ts:70]: AFAIK there is no difference. Just coding style :)

Comment (phated) [reply addons/docs/src/frameworks/vue/sourceDecorator.ts:70]: I'm fine with the current then. 👍

**Comment by phated** [comment]
Unsure why the e2e-tests-yarn-2-pnp are still failing 🤔

**Comment by pocka** [comment]
CI has been failing for some time span (?). I rebased the `next` branch and updated the PR.

**Pull request merged**